src/docs/arch-column-row.dox

```diff
@@ -0,0 +1,93 @@
+/*! @arch_page arch-column-row Row Store & Column Store
+
+@section Definition
+
+A row oriented storage is an access method where all columns of a row are stored
+together, the data is stored by row, while a column oriented system stores its data tables by
+column. WiredTiger supports both and each management system is represented internally by a certain
+type of B-Tree data structure.
+
+Keys and values in row-stores can be anything of arbitrary length, thus, row-stores implement a
```

Comment:
Technically, not "arbitrary", it's like `32bits - a small amount`

Reply:
@keithbostic , are keys for row-stores always made of 32 bits ?

Reply:
I may have misunderstood your comment.
I wanted to say here we could use any length for a key really. I thought that was the case. Is it restricted in terms of length ?

Reply:
It's restricted in length, it can be just under 4B bytes long, that is, they can go up to just under 4GB in length. Large, but not arbitrarily large.

Reply:
I went with the word `variable`, I hope it's fine. See 2ab4e82d6cb995d4704fd1a97018a9f17be6d18b